mapper/tedge_mapper/src/c8y_converter.rs

```diff
@@ -2,33 +2,91 @@ use crate::converter::*;
 use crate::error::*;
 use crate::size_threshold::SizeThreshold;
 use mqtt_client::{Message, Topic};
+use std::collections::HashSet;
+
+const SMARTREST_PUBLISH_TOPIC: &str = "c8y/s/us";
 
 pub struct CumulocityConverter {
     pub(crate) size_threshold: SizeThreshold,
+    children: HashSet<String>,
+    pub(crate) mapper_config: MapperConfig,
+}
+
+impl CumulocityConverter {
+    pub fn new(size_threshold: SizeThreshold) -> Self {
+        let mut topic_fiter = make_valid_topic_filter_or_panic("tedge/measurements");
+        let () = topic_fiter
+            .add("tedge/measurements/+")
+            .expect("invalid topic filter");
+
+        let mapper_config = MapperConfig {
+            in_topic_filter: topic_fiter,
+            out_topic: make_valid_topic_or_panic("c8y/measurement/measurements/create"),
+            errors_topic: make_valid_topic_or_panic("tedge/errors"),
+        };
+
+        let children: HashSet<String> = HashSet::new();
+        CumulocityConverter {
+            size_threshold,
+            children,
+            mapper_config,
+        }
+    }
 }
 
 impl Converter for CumulocityConverter {
     type Error = ConversionError;
-    fn convert(&self, input: &str) -> Result<String, Self::Error> {
-        let () = self.size_threshold.validate(input)?;
-        c8y_translator_lib::json::from_thin_edge_json(input).map_err(Into::into)
+
+    fn get_mapper_config(&self) -> &MapperConfig {
+        &self.mapper_config
     }
 
-    fn convert_child_device_payload(
-        &self,
-        input: &str,
-        child_id: &str,
-    ) -> Result<String, Self::Error> {
-        let () = self.size_threshold.validate(input)?;
-        c8y_translator_lib::json::from_thin_edge_json_with_child(input, child_id)
-            .map_err(Into::into)
+    fn convert_messages(&mut self, input: &Message) -> Result<Vec<Message>, ConversionError> {
+        let () = self.size_threshold.validate(input.payload_str()?)?;
+
+        let mut vec: Vec<Message> = Vec::new();
+
+        let maybe_child_id = get_child_id_from_topic(&input.topic.name)?;
+        match maybe_child_id {
+            Some(child_id) => {
+                let c8y_json_child_payload =
```

Comment:
We can't move it to prevent a bug: if user publishes invalid JSON, then publishes a valid JSON mesage, the mapper doesn't publish a child device creation message.

So, have to keep those order due to:
- We need to return an error when the input Thin Edge JSON is invalid before creating the child device creation message.
- Child device creation messages must be added to the vector before the measurement JSON message.

However, I can add a comment. Maybe people will wonder the order.